### PR TITLE
intro memory_mib_max resource field to OOM containers

### PR DESCRIPTION
**Pull request by thundergolfer**
## Describe your changes

Adds new field that is required to handle OOMing containers based on user requested limit. 

### Backward/forward compatibility checks

- [x] Client+Server: this change is compatible with old servers
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself

### CODE REVIEW

File: modal_proto/api.proto [hunk 1]
```diff
@@ -1456,9 +1456,10 @@ message RateLimit {
 }
 
 message Resources {
-  uint32 memory_mb = 2;
-  uint32 milli_cpu = 3;
+  uint32 memory_mb = 2; // MiB
+  uint32 milli_cpu = 3; // milli CPU cores
   GPUConfig gpu_config = 4;
+  uint32 memory_mib_max = 5; // MiB
```
Comment (mwaskom) [inline modal_proto/api.proto:1462]: YMMV but it feels arguably more error-prone to have inconsistent naming schemes than to just be consistently imprecise in the language.

Comment (mwaskom) [reply modal_proto/api.proto:1462]: Alternatively just bite the bullet on deprecating the _mb suffix so we can be internally and externally consistent at some point in the future.

Comment (thundergolfer) [reply modal_proto/api.proto:1462]: I think I can _maybe_ rename the field without much trouble because proto cares about field number not the name.

Comment (mwaskom) [reply modal_proto/api.proto:1462]: Oh yeah that’s smart. I’m not experienced enough with protobuf to say that is definitely a good idea but it does correspond to my understanding of how they work.

Comment (mwaskom) [reply modal_proto/api.proto:1462]: I guess the tricky part is coordinating the change between between the client / server codebases; you'll still need to start serverside and make it check both attribute names I expect.

Comment (thundergolfer) [reply modal_proto/api.proto:1462]: I don't have the time to do this rename carefully so I'll just go with your suggestion of the consistently imprecise naming :ok

**Review by mwaskom — APPROVED** [review]

**Pull request merged**